crates/kas-core/src/core/tile.rs

```diff
@@ -83,6 +83,27 @@ pub trait Tile: Layout {
         unimplemented!() // make rustdoc show that this is a provided method
     }
 
+    /// Describe the widget's role
+    ///
+    /// This descriptor supports accessibility tooling and UI introspection.
+    ///
+    /// The default implementation simply returns [`Role::Unknown`].
+    fn role(&self, cx: &mut dyn RoleCx) -> Role<'_> {
+        let _ = cx;
+        Role::Unknown
+    }
+
+    /// Specify additional role properties for child `index`
+    ///
+    /// Role properties (for example [`RoleCxExt::set_label`]) may be specified
+    /// by a widget for itself in [`Self::role`] or by a parent in this method.
+    ///
+    /// The default implementation does nothing.
+    #[inline]
+    fn role_child_properties(&self, cx: &mut dyn RoleCx, index: usize) {
+        let _ = (cx, index);
+    }
```

Comment:
This is a slightly strange design, but allows determination of any widget's role + role properties without recursing the whole tree.

(For finding those children, we already have `child_indices` and `get_child`.)